generators/entity-client/templates/react/src/main/webapp/app/entities/entity.tsx.ejs

```diff
@@ -36,34 +49,57 @@ import {
 } from './<%= entityFileName %>.reducer';
  // tslint:disable-next-line:no-unused-variable
 import { APP_DATE_FORMAT, APP_LOCAL_DATE_FORMAT } from 'app/config/constants';
+<%_ if (pagination !== 'no') { _%>
+import { ITEMS_PER_PAGE } from 'app/shared/util/pagination.constants';
+<%_ } _%>
 
 export interface I<%= entityReactName %>Props {
   getEntities: ICrudGetAllAction;
   <%_ if (searchEngine === 'elasticsearch') { _%>
   getSearchEntities: ICrudGetAllAction;
   <%_ } _%>
   <%=entityInstancePlural %>: any[];
+  <%_ if (pagination === 'pagination') { _%>
+  totalItems: 0;
+  location: any;
+  history: any;
+  <%_ } _%>
   match: any;
 }
-<%_ if (searchEngine === 'elasticsearch') { _%>
 
+<% if (searchEngine === 'elasticsearch' && pagination !== 'no') { _%>
+export interface I<%= entityReactName %>State extends IPaginationBaseState {
+  search: string;
+}
+<%_ } else if (searchEngine === 'elasticsearch') { _%>
 export interface I<%= entityReactName %>State {
   search: string;
 }
+<%_ } else if (pagination !== 'no') { _%>
+export type I<%= entityReactName %>State = IPaginationBaseState;
 <%_ } _%>
 
-export class <%= entityReactName %> extends React.Component<I<%= entityReactName %>Props<% if (searchEngine === 'elasticsearch') { %>, I<%= entityReactName %>State<% } %>> {
-<%_ if (searchEngine === 'elasticsearch') { _%>
+export class <%= entityReactName %> extends React.Component<I<%= entityReactName %>Props<% if (searchEngine === 'elasticsearch' || pagination !== 'no') { _%>, I<%= entityReactName %>State<%_ } _%>> {
+  <% if (searchEngine === 'elasticsearch' || pagination !== 'no') { _%>
```

Comment:
use newline template `<%_`

Reply:
change to 
```
<%_ if (searchEngine === 'elasticsearch' || pagination !== 'no') { _%>
  state: I<%= entityReactName %>State = {
    <%_ if (searchEngine === 'elasticsearch') { _%>
    search: '',
     <%_ } _%>
     <%_ if (pagination !== 'no') { _%>
     ...getSortState(this.props.location, ITEMS_PER_PAGE),
      <%_ } _%>
  }
 <%_ } _%>
```

dont worry about extra comma, prettier transform will take care of it